src/test/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/caseupdated/CaseUpdatedAboutToSubmitHandlerTest.java

```diff
@@ -679,6 +679,89 @@ public void givenHearingTypeOralAndWantsToAttendHearingNo_thenAddWarningMessage(
+        callback.getCaseDetails().getCaseData().getAppeal().getHearingOptions().setExcludeDates(excludeDate);
+
+
+        PreSubmitCallbackResponse<SscsCaseData> response = handler.handle(ABOUT_TO_SUBMIT, callback, USER_AUTHORISATION);
+        assertThat(response.getErrors().iterator().next(), is("Add a start date for unavailable dates"));
+    }
+
+    @Test
+    public void givenAnAppealWithIncorrectExcludedEndDate_thenProvideErrorMessage() {
+        List<ExcludeDate> excludeDate;
+        excludeDate = new ArrayList<>(List.of(
+                ExcludeDate.builder().value(DateRange.builder()
+                        .start("2023-06-17")
+                        .end(null)
+                        .build()).build()));
+
+        callback.getCaseDetails().getCaseData().getAppeal().setHearingOptions(HearingOptions.builder().wantsToAttend("Yes").scheduleHearing("Yes").build());
+        callback.getCaseDetails().getCaseData().getAppeal().getHearingOptions().setExcludeDates(excludeDate);
+
+
+        PreSubmitCallbackResponse<SscsCaseData> response = handler.handle(ABOUT_TO_SUBMIT, callback, USER_AUTHORISATION);
+        assertThat(response.getErrors().iterator().next(), is("Add a end date for unavailable dates"));
+    }
+
+    @Test
+    public void givenAnAppealWithEmptyExcludedDates_thenProvideErrorMessage() {
+        List<ExcludeDate> excludeDate;
+        excludeDate = new ArrayList<>(List.of(
+                ExcludeDate.builder().value(DateRange.builder()
+                        .start(null)
+                        .end(null)
+                        .build()).build()));
+
+        callback.getCaseDetails().getCaseData().getAppeal().setHearingOptions(HearingOptions.builder().wantsToAttend("Yes").scheduleHearing("Yes").build());
+        callback.getCaseDetails().getCaseData().getAppeal().getHearingOptions().setExcludeDates(excludeDate);
+
+
+        PreSubmitCallbackResponse<SscsCaseData> response = handler.handle(ABOUT_TO_SUBMIT, callback, USER_AUTHORISATION);
+        assertThat(response.getErrors().iterator().next(), is("Add a start date for unavailable dates"));
+    }
+
+    @Test
+    public void givenAnAppealWithCorrectExcludedDates_thenDontProvideError() {
+        List<ExcludeDate> excludeDate;
+        excludeDate = new ArrayList<>(List.of(
+                ExcludeDate.builder().value(DateRange.builder()
+                        .start("2023-06-17")
+                        .end("2023-06-19")
+                        .build()).build()));
+
+        callback.getCaseDetails().getCaseData().getAppeal().setHearingOptions(HearingOptions.builder().wantsToAttend("Yes").scheduleHearing("Yes").build());
+        callback.getCaseDetails().getCaseData().getAppeal().getHearingOptions().setExcludeDates(excludeDate);
+
+
+        PreSubmitCallbackResponse<SscsCaseData> response = handler.handle(ABOUT_TO_SUBMIT, callback, USER_AUTHORISATION);
+        assertThat(response.getErrors().isEmpty(), is(true));
```

Comment:
missing a few bits of CC:
https://sonarcloud.io/summary/new_code?id=SSCSTCA&pullRequest=3335